Make git.cmd.Git.CatFileContentStream iterable

Hex052 opened the pull request:
Add __next__ method to git.cmd.Git.CatFileContentStream, so it can actually be used as an iterable

Byron commented:
How does an iterator indicate 'end of iteration'? If I remember correctly, it needs to throw a special exception which would break the API of 'next()'.
If iteration should be implemented, that could of course be done but should not break existing APIs.
Maybe I am missing something though.

Hex052 commented:
To end iteration, an iterator needs to raise the exception [`StopIteration`](https://docs.python.org/3/library/exceptions.html#StopIteration), which is currently done in the `next()` method. To avoid breaking anything that might possibly use the existing `next()` method, I just added a new function that called the existing one. The existing function will only raise `StopIteration` at the very end (not on blank lines) since `readline()` does not strip any trailing newlines.

I added `__next__` because [python requires both `__iter__` and `__next__`](https://docs.python.org/3/library/stdtypes.html?highlight=__next__#iterator-types) in the iterator.

Hopefully that clears it up?

Byron commented:
Thanks a lot for filling in the blanks :) and for making this contribution!
